### ISSUE TITLE
Shareable link to a Gist returns 404

### ISSUE THREAD

**Issue by bzz** [bblfsh/dashboard#88]
On build from latest master, if one imports from Gist URL https://gist.github.com/bzz/cf2cc8d763bf5013f3a86d81bdb3ae8c, UAST is displayed, and "Share" btn produces a link

http://0.0.0.0:9999/bzz/cf2cc8d763bf5013f3a86d81bdb3ae8c/raw/284c8c6feec80428d1a1c6ff4cdba2c5799473c9/ParsingSucceeds.java

that, when pasted it to a new tab, results in `404 page not found` page.

**Comment by smacker** [comment]
oh. Thank you! I tested only in dev mode, most probably need to change routing in go server.

**Comment by smacker** [comment]
the link is broken in master now, because of new bug during refactoring. Fix: https://github.com/bblfsh/dashboard/pull/90
But even with wrong link it doesn't return me `404`.

Can you please re-test with that commit?

**Comment by bzz** [comment]
Cool! Will try again and get back to you asap.

**Comment by bzz** [comment]
@smacker could you let me know if I'm missing something, but after fresh build from master nothing has changed for me

```
git show                                                                                               
commit 085f8763c64285c3068f9be657e999ac57fbd1ea (HEAD -> master, upstream/master, upstream/HEAD, origin/master)
Merge: 23aa682 e09ca4e
Author: Maxim Sukharev <max@smacker.ru>
Date:   Fri Dec 15 21:18:23 2017 +0500
```

And the link above http://0.0.0.0:9999/bzz/cf2cc8d763bf5013f3a86d81bdb3ae8c/raw/284c8c6feec80428d1a1c6ff4cdba2c5799473c9/ParsingSucceeds.java

results in 404 as before

```
INFO[0003] [GIN] 2017/12/15 - 20:33:23 | 404 |         131ns |       127.0.0.1 |  GET     /bzz/cf2cc8d763bf5013f3a86d81bdb3ae8c/raw/284c8c6feec80428d1a1c6ff4cdba2c5799473c9/ParsingSucceeds.java
```

Does it work for you? 

Or what is the better way of testing link sharing use-case that you have used?

**Comment by smacker** [comment]
yes. It works for me. In chrome. Dashboard installed service worker and request don't hit server. ><
So my original thought was correct.
Fixed in https://github.com/bblfsh/dashboard/pull/98 
Tested in Safari&Incognito mode.

**Comment by bzz** [comment]
@smacker not sure what am I doing wrong:
```
git fetch upstream pull/98/head:fix/noroute
make clean
make serve
open http://0.0.0.0:9999/bzz/cf2cc8d763bf5013f3a86d81bdb3ae8c/raw/284c8c6feec80428d1a1c6ff4cdba2c5799473c9/ParsingSucceeds.java
```

and try different browsers (Safari, Chromium, FF) - this URL always results in 404 for me :/

**Comment by smacker** [comment]
it's weird. In the branch, I set another `index.html` as 404. So server shouldn't return regular 404 at all.
btw `0.0.0.0` doesn't route to the server on my system, I tested with `127.0.0.1`

**Comment by bzz** [comment]
I see! Well, I just have copied the output URL that `make serve` pointed me to.

Let me try again with 127.0.0.1 and attach DevTools finds

**Comment by bzz** [comment]
Ok, http://127.0.0.1:9999/bzz/cf2cc8d763bf5013f3a86d81bdb3ae8c/raw/284c8c6feec80428d1a1c6ff4cdba2c5799473c9/ParsingSucceeds.java actually does work as expected!

**Comment by smacker** [comment]
merged
